src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
```

Comment:
[CODE PERFORMANCE] Directly modifying the `Animal.alive` array within the `applyDamage` method by calling `cleanupDeadAnimals` creates a dependency that could lead to performance issues when there are nested loops. Since `cleanupDeadAnimals` is already filtering the `Animal.alive` array to remove dead animals, consider calling this cleanup method separately at a point in the code where it makes sense, such as after all the damage has been applied in a turn, to avoid potential nested loop performance problems.